src/integrations/prefect-aws/prefect_aws/experimental/bundles/execute.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import os
+import tempfile
+from pathlib import Path
+
+import typer
+from botocore.exceptions import ClientError
+from pydantic_core import from_json
+
+from prefect.runner import Runner
+from prefect.utilities.asyncutils import run_coro_as_sync
+from prefect_aws.credentials import AwsCredentials
+
+from .types import AwsCredentialsBlockName, S3Bucket, S3Key
+
+
+def download_bundle_from_s3(
+    bucket: S3Bucket,
+    key: S3Key,
+    output_dir: str | None = None,
+    aws_credentials_block_name: AwsCredentialsBlockName | None = None,
+) -> dict[str, str]:
+    """
+    Downloads a bundle from an S3 bucket.
+
+    Args:
+        bucket: S3 bucket name
+        key: S3 object key
+        output_dir: Local directory to save the bundle (if None, uses a temp directory)
+        aws_credentials_block_name: Name of the AWS credentials block to use
```

Comment:
two notes:
- worth calling out if block is `None`, credentials inferred from environment
- prob worth doc'ing the return value (what is generally in the dict?)